### PR TITLE
Improve error output and traces in the ank CLI

### PR DESCRIPTION
**Pull request by maturar**
Issues: #49

# Definition of Done

The PR shall be merged **only if all of following items were checked.** In case an item is not applicable as described, please provide a short explanation right after the item in question. Separate the explanation by a semicolon and write it as bold text like **; requirements are handled in issue #xyz**:

- [x] documentation of all modules `/*/doc/swdesign` is up-to-date
- [x] requirements are up-to-date and requirements for new features have been added and mapped to source and test
- [x] conform to [unit verification strategy](https://eclipse-ankaios.github.io/ankaios/main/development/unit-verification/)

### CODE REVIEW

**Comment by maturar** [comment]
The implementation is done. The ticket needs testing now.
Regarding documentation and requirements:
* We need to write a design decision in `swdesign`.
* We shall write a requirement which describes the concept. Not too detailed. It would be enough to write one requirement with a table which describes which functions shall do what.

**Comment by maturar** [comment]
The task has been implemented, the pull request is ready for review.

File: ank/src/log.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+/// This macro prints the message as a debug trace, if the CLI command is called with `--verbose` flag.
+/// If the CLI command is called without the `-verbose` flag, the macro does nothing.
+/// Calling this macro does not terminate the application.
+#[macro_export]
+macro_rules! output_debug {
+    ( $ ( $ arg : tt ) * ) => { $crate::log::output_debug_fn ( format_args ! ( $ ( $ arg ) * ) ) }
+}
+
+pub(crate) fn output_and_error_fn(args: fmt::Arguments<'_>) {
+    eprintln!("\x1b[31m\x1b[1merror:\x1b[0m {}", args);
```
[the remark this answers is not in the document]
Comment (maturar) [reply ank/src/log.rs:27]: This change has been made intentionally. The user may get errors from two sources. One source is `clap` and the second source are our tracing macros. I have changed format of our tracing to match to the tracing from the `clap`. So the user will always get the traces in the same format. Try to change this line and test the change with these commands for example:

1. `./ank get -verbose`
2. `./ank get workloads` (without running server, so it fails with an error).

If you have idea how to handle this situation, I am open for any suggestion :-)

File: ank/src/log.rs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+pub(crate) fn output_and_exit_fn(args: fmt::Arguments<'_>) {
+    println!("{}", args);
+    exit(0);
+}
+
+pub(crate) fn output_debug_fn(args: fmt::Arguments<'_>) {
+    if is_verbose() {
+        println!("\x1b[94mdebug:\x1b[0m {}", args);
```
[the remark this answers is not in the document]
Comment (maturar) [reply ank/src/log.rs:38]: The change has been made to keep this consistent with the error traces. See the previous comment. We have to discuss both cases together of course.

File: ank/src/log.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (inf17101) [reply ank/src/log.rs:27]: Ok, thank you, then it looks fine for me.

File: ank/src/log.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (inf17101) [reply ank/src/log.rs:38]: Same like above, LGTM :-)

**Review by inf17101 — APPROVED** [review]
LGTM 👍

**Pull request merged**